### PR TITLE
ISPN-12949 Remove the obsolete IndexQueryMode

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.redhat.com/browse/ISPN-12949

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/upgrading.adoc [hunk 1]
```diff
@@ -1,5 +1,10 @@
 = Upgrading from 12.0 to 12.1
 
+== Search
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/upgrading.adoc:3]: should be a new section "Upgrading from 12.1 to 13.0"

File: documentation/src/main/asciidoc/topics/upgrading.adoc [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
+
+The optional ```IndexedQueryMode``` parameter has been removed from queries in library mode, CLI, Hot Rod and REST.
```
Comment (pruivo) [inline documentation/src/main/asciidoc/topics/upgrading.adoc:5]: shouldn't `IndexedQueryMode` has been deprecated first?

File: documentation/src/main/asciidoc/topics/upgrading.adoc (hunk 1 above)
Comment (gustavonalle) [reply documentation/src/main/asciidoc/topics/upgrading.adoc:3]: Yes, I think I've sent the PR before 13.0.0-SNAPSHOT

File: documentation/src/main/asciidoc/topics/upgrading.adoc (hunk 2 above)
Comment (gustavonalle) [reply documentation/src/main/asciidoc/topics/upgrading.adoc:5]: Ideally yes, but since 11 or 12 it's being ignored internally

**Review by pruivo — APPROVED** [review]

**Review by karesti — APPROVED** [review]

**Comment by pruivo** [comment]
integrated! thanks @gustavonalle !

**Pull request merged**